src/storm/mesos/logviewer/LogViewerController.java

```diff
@@ -0,0 +1,210 @@
+        // Decide on (b)
+        
+        // Testing indicates if the supervisor goes down the logviewer goes down.
+        // So this may never be testable. Of course this was using Vagrant so not 
+        // sure how valid the observation is.
+        
+        // Another approach would be to add a "ping" and "shutdown" rest service to 
+        // the log viewer directly, then we would simply ping it, and stop it.
+        
+        // How to determine the pid of a logviewer started by a now defunct supervisor?
+        
+        // For now just terminate this attempt at creating a logviewer.
+        throw new RuntimeException("Failed to terminate existing logviewer, will not start a new one.");
+       
+    }
+    
+    public void setUrlDetector(IUrlDetection urlDetector) {
+        this.urlDetector = urlDetector;
+    }
+    
+    public IUrlDetection getUrlDetector() {
+        return urlDetector;
+    }
+    
+    public void setInstallLocation(String location) {
+        installLocation = location;
+    }
+    
+    public String getInstallLocation() {
+        return installLocation;
+    }
+    
+    protected Process process;
+    protected IUrlDetection urlDetector;
+    protected String installLocation;
+    protected Map<String,Object> config;
+    protected Optional<Integer> port;
+    protected static final String LOG_VIEWER_PORT = "logviewer.port";
+       
+    protected void setProcess(Process process) {
+        this.process = process;
+    }
+    
+    protected Process getProcess() {
+        return process;
+    }
+    
+    protected void setPort(Optional<Integer> oport) {
+        port = oport;
+    }
+    
+    protected Optional<Integer> getPort() {
+        return port;
+    }
+    
+    protected void setConfig(Map<String,Object> configuration) {
+        config = configuration;
+    }
+    
+    protected Map<String,Object> getConfig(){
+        return config;
+    }
+    
+    protected void launchLogViewer () throws IOException {
+         
+        if(LOG.isDebugEnabled())
+            LOG.debug("launchLogViewer: home dir? " + getInstallLocation());
+        
+        ProcessBuilder pb = createProcessBuilder(getInstallLocation());
+        setProcess(pb.start());       
+    }
+    
+    /**
+     * Create a process builder to launch the log viewer
+     * @param logDirectory
+     * @return
+     */
+    protected ProcessBuilder createProcessBuilder(String homeDirectory) {
+        ProcessBuilder pb = new ProcessBuilder(homeDirectory + "/bin/storm","logviewer");
+         
+        // If anything goes wrong at startup we want to see it.
+        File log = new File(homeDirectory + "/logs/logviewer-startup.log");
+        pb.redirectErrorStream(true);
+        pb.redirectOutput(Redirect.appendTo(log)); 
+        return pb;
+    }
+    
+    @SuppressWarnings("unchecked")
+    protected void initialize() {
+        setInstallLocation(new File("./").getAbsolutePath());
```

Comment:
`new File("./").getAbsolutePath()` seems like a weird way to get the current working dir. Can you do this in a more idiomatic way?